corporate/models.py

```diff
@@ -30,9 +36,27 @@ class Customer(models.Model):
     # they purchased.
     exempt_from_from_license_number_check: bool = models.BooleanField(default=False)
 
+    @property
+    def is_self_hosted(self) -> bool:
+        return self.remote_server is not None and self.realm is None
```

Comment:
I think we don't want an `and` gate here.  If you want an assertion that the invariants are correct, one can do this:

```
is_self_hosted = self.remote_server is not None
if is_self_hosted:
     assert self.realm is None
return is_self_hosted
```

Reply:
This sort of constraint is probably better done with https://adamj.eu/tech/2020/03/25/django-check-constraints-one-field-set/ ; can you look at that as a follow-up cleanup?